src/dataExplorer/context/persistance.tsx

```diff
@@ -1,43 +1,57 @@
-import React, {FC, createContext} from 'react'
-import {TimeRange} from 'src/types'
+import React, {FC, createContext, useCallback} from 'react'
+import {TimeRange, RecursivePartial} from 'src/types'
 import {DEFAULT_TIME_RANGE} from 'src/shared/constants/timeRanges'
 import {useSessionStorage} from 'src/dataExplorer/shared/utils'
 import {Bucket} from 'src/types'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
 
-interface MeasurementSelection {
+interface SchemaComposition {
+  synced: boolean
+  diverged: boolean
+}
+export interface SchemaSelection {
   bucket: Bucket
   measurement: string
+  fields: string[]
+  composition: SchemaComposition
 }
 
 interface ContextType {
   horizontal: number[]
   vertical: number[]
   range: TimeRange
   query: string
-  selection: MeasurementSelection
+  selection: SchemaSelection
 
   setHorizontal: (val: number[]) => void
   setVertical: (val: number[]) => void
   setRange: (val: TimeRange) => void
   setQuery: (val: string) => void
-  setSelection: (val: MeasurementSelection) => void
+  setSelection: (val: RecursivePartial<SchemaSelection>) => void
```

Comment:
Why `RecursivePartial<SchemaSelection>` instead of just `SchemaSelection`? From my understanding, `RecursivePartial` is for recursive data structure, but `SchemaSelection` has a set data structure instead of recursive.

Reply:
The `setSchemaSelection()` method in the persistence context takes in this recursive partial. Meaning:
* we have a nested object
* we want to only update part of the nested object
* therefore, the method is updated such that we pass in only the nested part...and the `setSchemaSelection()` updates only that nested prop.

What we have been doing is:
* Passing in an object which is: `setSelection({...old, ...new})`, at the level of many different components.
* In order to make that^^ work, we have a code pattern where we pass the old/previous session object around to these components.
* This works...until you hit possible points where we are not memoizing correctly.

Since we plan to eventually use the new ScriptBuilder everywhere, (which will include many other devs using memorization), it felt prudent to defensively encode against this exposure to memoziation. That is why I updated the `PersistenceContext.setSchemaSelection()` to take it's own state as the `...old`, and we only pass in the partial recursive object for the `...new`.

Does this make sense?

Reply:
Got it. Thanks for explaining!